sync2/client.go

```diff
@@ -39,16 +41,30 @@ type HTTPClient struct {
 }
 
 func NewHTTPClient(shortTimeout, longTimeout time.Duration, destHomeServer string) *HTTPClient {
+	baseUrl := destHomeServer
+	if strings.HasPrefix(destHomeServer, "/") {
+	     baseUrl = "http://unix"
+	}
+
 	return &HTTPClient{
-		LongTimeoutClient: &http.Client{
-			Timeout:   longTimeout,
-			Transport: otelhttp.NewTransport(http.DefaultTransport),
-		},
-		Client: &http.Client{
-			Timeout:   shortTimeout,
-			Transport: otelhttp.NewTransport(http.DefaultTransport),
-		},
-		DestinationServer: destHomeServer,
+		LongTimeoutClient: newClient(longTimeout, destHomeServer),
+		Client:            newClient(shortTimeout, destHomeServer),
+		DestinationServer: baseUrl,
+	}
+}
+
+func newClient(timeout time.Duration, destHomeServer string) *http.Client {
+	transport := http.DefaultTransport
+	if strings.HasPrefix(destHomeServer, "/") {
```

Comment:
Fairly sure unix sockets can also be relative file paths, so this isn't ideal (albeit it is convenient). I would definitely split this out to a helper function in case we want to be a bit more intelligent around unix socket detection. E.g something in `internal/util.go`:
```go
func isUnixSocket(httpOrUnixStr string) bool {
    return strings.HasPrefix(httpOrUnixStr, "/")
}
```

It might be better to just say `strings.HasPrefix("http")`? Not really sure.

Reply:
extracted HomeServerUrl is it better?